gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveDatasetFinder.java

```diff
@@ -294,6 +304,12 @@ protected HiveDataset computeNext() {
     };
   }
 
+  protected static boolean shouldAllowTableLocation(Optional<String> regex, Table table) {
+    if (!regex.isPresent()) {
+      return true;
+    }
+    return Pattern.compile(regex.get()).matcher(table.getSd().getLocation()).matches();
```

Comment:
Pattern.compile(regex.get()) is called every time the method shouldAllowTableLocation is executed when the regex is present.  Compiling a regex every time can be inefficient, we can compile the pattern once and store it, eg.:
`private static Pattern compiledAllowlistPattern = regex.map(Pattern::compile).orElse(null);`

Reply:
I compiled it as part of the class field instead, good callout